USER_GUIDE.md

```diff
@@ -0,0 +1,206 @@
+# User Guide: Running HuggingFace LLaMA 2 Training on v4 and v5e
+
+
+This user guide provides a concise overview of the essential steps required to run HuggingFace (HF) Llama 2 training on both v4 and v5e. It first demonstrates how to do so on barebone TPU VMs, and then shares a Dockerfile for those who would prefer to run it in a container.
+
+
+## Environment Setup
+
+The following setup assumes to run the training job with Llama 2 7B.
+
+
+### Cloud TPU VM Creation
+```
+export TPU_NAME=jwtan-v4-32
+export ZONE=us-central2-b
+export ACCELERATOR_TYPE=v4-32
+export RUNTIME_VERSION=tpu-ubuntu2204-base
+export PROJECT=mlperf-high-priority-project
+
+gcloud alpha compute tpus queued-resources create $TPU_NAME --node-id $TPU_NAME --zone=$ZONE --project=$PROJECT --accelerator-type=$ACCELERATOR_TYPE --runtime-version=$RUNTIME_VERSION
+```
+
+Change `{TPU\_NAME, ZONE, PROJECT, RUNTIME\_VERSION}` as needed. For v5e, use the `v2-alpha-tpuv5-lite` runtime version in a supported project and zone. And use the following command to query the status of the creation request:
+```
+gcloud alpha compute tpus queued-resources describe ${TPU_NAME}  --project ${PROJECT} --zone ${ZONE}
+```
+
+Once the QueuedResource’s status becomes ACTIVE, the TPU VMs are provisioned and ready to run your training workload. See the [official guide](https://cloud.google.com/tpu/docs/queued-resources) for more information on a QueuedResource’s life cycle.
+
+
+### HF Llama 2 Environment Setup
+
+Here PyTorch 2.1 and PyTorch/XLA 2.1 are used with our fork of HuggingFace. At the time where this doc was created, the 2.1 releases were not officially out yet, and therefore, pre-releases links are used.
+```
+gcloud alpha compute tpus tpu-vm ssh ${TPU_NAME} \
+--zone ${ZONE} \
+--project ${PROJECT} \
+--worker=all \
+--command='
+# Step 1: install torch, torch-xla, libtpu
+pip3 install torch~=2.1.0 torch_xla[tpu]~=2.1.0rc5 -f https://storage.googleapis.com/libtpu-releases/index.html
+
+# Step 2: install HF
+git clone -b llama2-google-next-training https://github.com/pytorch-tpu/transformers.git
+cd transformers
+sudo pip3 install -e . --user
+pip3 install datasets accelerate evaluate scikit-learn
+'
+```
+
+The last step for HF setup is to copy your Llama 2 config into the TPU VM.
+```
+gcloud compute tpus tpu-vm scp llama_2_7B.config $TPU_NAME:~/config.json --worker all --project $PROJECT --zone=$ZONE
+```
+
+
+## Steps to Run HF LLaMA
+```
```

Comment:
nit: Can we add a sentence here so it doesn't jump straight from the heading to code?

Reply:
Done.